src/components/AppMap.ts

```diff
@@ -319,6 +320,35 @@ export default class AppMap extends mixins(MixinUtil) {
     this.sidebar.open(pane);
   }
 
+  static addGeoJSONFeatureToLayer(layer: any) {
+    if (!layer.feature) {
+      layer.feature = {type: 'Feature', properties: {title: "", text: ""}};
+    }
+  }
+
+  addPopupAndTooltip(layer: L.Marker | L.Polyline) {
```

Comment:
Can't this be static as well? Come to think of it, I think these two functions (addGeoJSONFeatureToLayer and addPopupAndTooltip) should probably be "free functions" (not part of this class at all, like getMarkerDetailsComponent)